src/System.Linq.Expressions/src/System/Linq/Expressions/IndexExpression.cs

```diff
@@ -272,14 +272,13 @@ private static string GetArgTypesString(Expression[] arguments)
 
         private static PropertyInfo FindProperty(Type type, string propertyName, Expression[] arguments, BindingFlags flags)
         {
-            var props = type.GetProperties(flags).Where(x => x.Name.Equals(propertyName, StringComparison.CurrentCultureIgnoreCase)); ;
-            PropertyInfo[] members = new List<PropertyInfo>(props).ToArray();
-            if (members == null || members.Length == 0)
+            var members = type.GetProperties(flags)
+                .Where(x => x.Name.Equals(propertyName, StringComparison.CurrentCultureIgnoreCase));
+            if (!members.Any())
```

Comment:
You're correct, I think I understand what you're saying (`GetEnumerator` simply returns the same object for iterator state machines, but needs to allocate a new one if it's called twice). I'll remove the `Any` call.